### PR TITLE
Feature/send token

### PR DESCRIPTION
**Pull request by Yamaguchi**
- Create a transaction to send token in wallet to other address
- Sign the transaction by private key stored in wallet.

### CODE REVIEW

File: ts_src/token.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
+import * as tapyrus from 'tapyrusjs-lib';
+import { Utxo } from './utxo';
+import Wallet from './wallet';
+
+export interface Token {
+  transfer(
```
Comment (azuchi) [inline ts_src/token.ts:6]: Wouldn't it be better for wallet to implement the `transfer`?

File: ts_src/token.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  }> {
+    const txb = new tapyrus.TransactionBuilder();
+    txb.setVersion(1);
+
+    const inputs: Utxo[] = [];
+
+    const uncoloredScript = tapyrus.payments.p2pkh({
+      output: changePubkeyScript,
+    });
+    params.forEach(async param => {
```
Comment (azuchi) [inline ts_src/token.ts:53]:
```suggestion
    for (const param of params) {
```

File: ts_src/signer.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
+import * as tapyrus from 'tapyrusjs-lib';
+import { Utxo } from './utxo';
+import Wallet from './wallet';
+
+export class SignOptions {
+  network?: tapyrus.Network;
+}
+
+export async function sign(
```
Comment (azuchi) [inline ts_src/signer.ts:9]: This function does not used.

**Review by azuchi — CHANGES_REQUESTED** [review]

File: ts_src/signer.ts (hunk 3 above)
Comment (Yamaguchi) [reply ts_src/signer.ts:9]: Yes, This function is intended to be used by application code.
The following code is a sample code of transferring tokens.

```javascript
async function onPurchase() {
    const colorId = Buffer.from("....", "hex"); 
    const toAddress = "....";
    const amount = 1;

    const keys = await wallet.keyStore.keys();
    const pubkey = tapyrus.ECPair.fromPrivateKey(Buffer.from(keys[0], "hex")).publicKey!;
    const changePubkeyScript = tapyrus.payments.p2pkh({ pubkey: pubkey }).output!;

    const token = new wallet.BaseToken();

    // Create Transaction(Unsigned)
    const result = await token.transfer(myWallet, {colorId, toAddress, amount}, changePubkeyScript); 

    //Sign
    const signedTxb = await wallet.signer.sign(myWallet, result.txb, result.inputs);

    // Broadcast
    await myWallet.broadcast(signedTxb.build()); 
}
```

File: ts_src/token.ts (hunk 2 above)
Comment (Yamaguchi) [reply ts_src/token.ts]: 6862ab1

File: ts_src/token.ts (hunk 1 above)
Comment (Yamaguchi) [reply ts_src/token.ts]: Updated at e1e3509

File: types/wallet.d.ts [hunk 4]
```diff
@@ -12,6 +15,12 @@ export default interface Wallet {
     update(): Promise<void>;
     broadcast(tx: tapyrus.Transaction): Promise<string>;
     balance(colorId?: string): Promise<Balance>;
+    utxos(colorId?: string): Promise<Utxo[]>;
+    estimatedFee(tx: tapyrus.Transaction): number;
+    transfer(params: TransferParams[], changePubkeyScript: Buffer): Promise<{
```
Comment (azuchi) [inline types/wallet.d.ts:20]: Isn't it better for transfer not only to build a remittance transaction but also to sign and broadcast it?

Comment (Yamaguchi) [reply types/wallet.d.ts:20]: I'll try to fix.

Comment (Yamaguchi) [reply types/wallet.d.ts]: Yes. Updated at 2bb76eb

**Review by azuchi — APPROVED** [review]

**Pull request merged**